### ISSUE TITLE
Regex for patternProperties number is incorrect

### ISSUE THREAD

**Issue by TrevorSundberg** [YousefED/typescript-json-schema#352]
```ts
interface Something {
  [value: number]: any;
}
```
Generates
```json
"patternProperties": {
  "^[0-9.]+$": {
    ...
  }
}
```

The regex `^[0-9.]+$` does not account for decimals or other valid values of numbers.

**Comment by piglovesyou** [comment]
I'm seeing `^[0-9]+$` in `typescript-json-schema@0.43.0`. So which case exactly are you concerned about?

**Comment by TrevorSundberg** [comment]
If you try use a number like 0.5 it fails, even though it's valid in typescript. Anything with decimals or scientific notation fails.

**Comment by jackbuehner** [comment]
@piglovesyou That's correct! I ran into an issue with the regex and your recommended change fixed it.
